### ISSUE TITLE
How to change PWM Frequency?

### ISSUE THREAD

**Issue by Ahmed-Mohiuddin-Shah** [bebump/coolth#2]
Is there a way to change to PWM frequency so that my fans don`t make a whining noise.
At slow RPM they create a whining noise which goes away when they are at full speed.
Is there a way where I can change the PWM frequency through the Arduino sketch?

**Comment by bebump** [comment]
The PWM frequency is determined by applying various dividers to the 16 MHz clock rate. You can change these divider values in the sketch.

It's worth noting that some of these dividers will also affect the time measurement, so it can mess with the RPM measurement functionality.

You can experiment with it though, just remember, that modifying some of the dividers will make the software display incorrect RPM values, and there isn't an easy way around it. It won't affect the behavior of the fans though, so your fans will be controlled correctly, it will be just the displayed values that may be incorrect.

https://www.etechnophiles.com/how-to-change-the-pwm-frequency-of-arduino-nano/

E.g. adding the line `TCCR2B = TCCR2B & B11111000 | B00000001;` somewhere in the beginning of the sketch will raise some PWM frequencies to 31 kHz.

That's as much as I remember from years back. I am using high quality PWM fans which aren't affected by the PWM signal's frequency, because they generate the motor driving signals internally, so I just left things on default in the Arduino sketch.

**Comment by Ahmed-Mohiuddin-Shah** [comment]
Thank you. The whine is gone.
I just added the following two line in the setup to change the PWM frequency for pins 3, 5, 6. 

TCCR2B = TCCR2B & B11111000 | B00000001;
TCCR0B = TCCR0B & B11111000 | B00000001;

![image](https://user-images.githubusercontent.com/83568426/126905997-2833d342-fefd-4c92-b319-44b52e6b139a.png)

I am using pins 3, 5, 6 as those are the pins my fans are attached to. My fans are two wire DC fans so they whine with the default PWM.

If some one else is having Whining Issue and is using the default sketch and schematic then add the following two lines:

TCCR2B = TCCR2B & B11111000 | B00000001;
TCCR1B = TCCR1B & B11111000 | B00000001;

![image](https://user-images.githubusercontent.com/83568426/126906182-43686c8a-e769-43ae-b416-4dfbde2494d0.png)

Thank you for your help.

**Comment by Ahmed-Mohiuddin-Shah** [comment]
UPDATE:
The program was having trouble connecting to the Arduino after adding the lines.

I have now changed the location of the line and the problem is not occurring now.

I removed the following two line in the setup and added them in void loop() before the analogWrite to change the PWM frequency for pins 3, 5, 6.

TCCR2B = TCCR2B & B11111000 | B00000001;
TCCR0B = TCCR0B & B11111000 | B00000001;
![image](https://user-images.githubusercontent.com/83568426/127122793-68e90b67-6fb7-4bba-868f-f05594a09f5a.png)

If some one is using the default sketch and schematic then add the following two lines:

TCCR2B = TCCR2B & B11111000 | B00000001;
TCCR1B = TCCR1B & B11111000 | B00000001;
![image](https://user-images.githubusercontent.com/83568426/127122586-504d59e5-86fc-4e79-95b7-f8d401b278eb.png)

I will update in two days if there are any other issues. If there are none the i will close the issue.
